debug/org.eclipse.debug.ui/ui/org/eclipse/debug/internal/ui/launchConfigurations/LaunchConfigurationTabGroupViewer.java

```diff
@@ -1455,7 +1455,7 @@ protected void handleTabSelected() {
 		fCurrentTabIndex = fTabFolder.getSelectionIndex();
 
 		ILaunchConfigurationTab[] tabs = getTabs();
-		if (previousTabIndex == fCurrentTabIndex || tabs == null || tabs.length == 0
+		if (tabs == null || tabs.length == 0
```

Comment:
There is the public API `setActiveTab()`. If that is ever called with the index of the already active tab, then your code change will first deactivate and then re-active that tab, which is probably not good.

Reply:
If I had to guess, I'd say the reason that this `if` returns `true` (_i.e._ the reason why `previousTabIndex == fCurrentTabIndex`) is because the change introduced in 5ca254b14421e64b122eed111b35621a1ef8e144 (which got reverted in 2a51e8c925354594e42fc758b332336db937e01c) is in direct conflict with the call in line 1680 of (`fTabFolder.setSelection(index);`)

And I agree with @tomaswolf : this change will have unintended (bad) implications

Reply:
> There is the public API `setActiveTab()`. If that is ever called with the index of the already active tab, then your code change will first deactivate and then re-active that tab, which is probably not good.

Now that previousTabIndex and fCurrentTabIndex are same, that condition is updated. So accordingly when they are not same only we need to call the deactivation followed by activation. 

Updated as per your review comments, Thanks for your review.

Reply:
If I apply this PR then calling `setActiveTab(1)` twice would activate the 2nd tab twice without deactivating it in-between. IMO also not good

Also, if I apply this PR and re-apply https://github.com/eclipse-platform/eclipse.platform.swt/pull/1044 then `testOnlyDefaultTabInOtherConfigIsActivated` breaks

I propose we define the expected behavior of `LaunchConfigurationTabGroupViewer` and its `setActiveTab(...)` and go from there. Once this is done, I can write some tests to check this behavior.

I'll start with:
1. All tabs are initialized upon creation of a `LaunchConfigurationTabGroupViewer`
2. The default tab is also activated upon creation
3. The default tab may come from the `CTabFolder`. If it does, it should be honored. If it doesn't, the 1st tab (index `0`) is the default tab
4. Taking the default tab from the `CTabFolder` should **not** lead to unnecessary activation/deactivation/initialization of tabs
5. Upon creation of a `LaunchConfigurationTabGroupViewer` only the following should occur
* All tabs are initialized (exactly once)
* The default tab (whether coming from `CTabFolder` or not) is activated exactly once
* No tab is deactivated
6. Activating the already active tab (_i.e._ calling `setActiveTab(0)` after creation) should neither deactivate the tab nor activate it again

Points 1, 2, 5 and 6 already hold before applying this PR + https://github.com/eclipse-platform/eclipse.platform.swt/pull/1044 (tests are missing for 5 and 6 though)
Points 3 and 4 should not break the other ones.

WDYT?

Reply:
Fixed by https://github.com/eclipse-platform/eclipse.platform/pull/1346